src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Utils.cs

```diff
@@ -756,9 +756,9 @@ internal static bool IsSelfSigned(X509Chain chain)
             return false;
         }
 
-        internal static AsymmetricAlgorithm GetAnyPublicKey(X509Certificate2 certificate)
+        internal static AsymmetricAlgorithm? GetAnyPublicKey(X509Certificate2 certificate)
         {
-            AsymmetricAlgorithm algorithm = (AsymmetricAlgorithm)certificate.GetRSAPublicKey() ?? certificate.GetECDsaPublicKey();
+            AsymmetricAlgorithm? algorithm = (AsymmetricAlgorithm)certificate.GetRSAPublicKey()! ?? certificate.GetECDsaPublicKey();
```

Comment:
Suffix-`!` followed by `??` makes no sense.  Perhaps the correct line is

```suggestion
            AsymmetricAlgorithm? algorithm = (AsymmetricAlgorithm?)certificate.GetRSAPublicKey() ?? certificate.GetECDsaPublicKey();
```

Reply:
I don't know what I was thinking there!